src/command_modules/azure-cli-keyvault/azure/cli/command_modules/keyvault/custom.py

```diff
@@ -531,7 +531,10 @@ def download_secret(client, vault_base_url, secret_name, file_path, encoding=Non
         raise CLIError("File or directory named '{}' already exists.".format(file_path))
 
     secret = client.get_secret(vault_base_url, secret_name, secret_version)
-    encoding = encoding or secret.tags.get('file-encoding', 'utf-8')
+
+    if not encoding:
+        encoding = secret.tags.get('file-encoding', 'utf-8') if secret.tags else 'utf-8'
```

Comment:
Is the tag added by service or it is a task for the user to add this tag?

Reply:
The CLI command `secret set` will determine and set this tag automatically. The user could also set it manually.  It a convenience for CLI 2.0 users. Otherwise it will try to download assuming UTF-8 unless the user supplies the `--encoding` argument.

From the help:
```
   --encoding -e          : Encoding of the destination file. By default, will look for the 'file-
                             encoding' tag on the secret. Otherwise will assume 'utf-8'.  Allowed
                             values: ascii, base64, hex, utf-16be, utf-16le, utf-8.
```